www/js/control/ProfileSettings.jsx

```diff
@@ -150,54 +255,206 @@ const ProfileSettings = () => {
+    async function forceSync() {
+        ClientStats.addEvent(ClientStats.getStatKeys().BUTTON_FORCE_SYNC).then(
+            function() {
+                console.log("Added "+ClientStats.getStatKeys().BUTTON_FORCE_SYNC+" event");
             });
+        ControlSyncHelper.forceSync().then(function() {
+            /*
+             * Change to sensorKey to "background/location" after fixing issues
+             * with getLastSensorData and getLastMessages in the usercache
+             * See https://github.com/e-mission/e-mission-phone/issues/279 for details
+             */
+            var sensorKey = "statemachine/transition";
+            return window.cordova.plugins.BEMUserCache.getAllMessages(sensorKey, true);
+        }).then(function(sensorDataList) {
+            Logger.log("sensorDataList = "+JSON.stringify(sensorDataList));
+            // If everything has been pushed, we should
+            // only have one entry for the battery, which is the one that was
+            // inserted on the last successful push.
+            var isTripEnd = function(entry) {
+                if (entry.metadata.key == getEndTransitionKey()) {
+                    return true;
+                } else {
+                    return false;
+                }
+            };
+            var syncLaunchedCalls = sensorDataList.filter(isTripEnd);
+            var syncPending = (syncLaunchedCalls.length > 0);
+            Logger.log("sensorDataList.length = "+sensorDataList.length+
+                       ", syncLaunchedCalls.length = "+syncLaunchedCalls.length+
+                       ", syncPending? = "+syncPending);
+            return syncPending;
+        }).then(function(syncPending) {
+            Logger.log("sync launched = "+syncPending);
+            if (syncPending) {
+                Logger.log("data is pending, showing confirm dialog");
+                setDataPendingVis(true); //consent handling in modal
+            } else {
+                setDataPushedVis(true);
+            }
+        }).catch(function(error) {
+            Logger.displayError("Error while forcing sync", error);
+        });
+    };
+
+    async function invalidateCache() {
+        window.cordova.plugins.BEMUserCache.invalidateAllCache().then(function(result) {
+            console.log("invalidate result", result);
+            setCacheResult(result);
+            setInvalidateSuccessVis(true);
+        }, function(error) {
+            Logger.displayError("while invalidating cache, error->", error);
+        });
     }
 
-    //conditional creation of setting sections
-    let userDataSection;
-    if(userDataSaved())
-    {
-        userDataSection = <ExpansionSection sectionTitle="control.user-data">
-                            <SettingRow textKey="control.erase-data" iconName="delete-forever" action={eraseUserData}></SettingRow>
-                            <ControlDataTable controlData={userData}></ControlDataTable>
-                        </ExpansionSection>;
```

Comment:
removed userData section in anticipation of eliminating calories from dashboard, and thus need to collect/maintain/show this data